data/scripts/actions/skinning.lua

```diff
@@ -0,0 +1,162 @@
+		[11269] = {chance = 10000, newItem = 5876, after = 11270}, -- lizard high guard
+		[11272] = {chance = 10000, newItem = 5876, after = 11270}, -- lizard high guard, after being killed
+		[11281] = {chance = 10000, newItem = 5876, after = 11282}, -- lizard zaogun
+		[11284] = {chance = 10000, newItem = 5876, after = 11282}, -- lizard zaogun, after being killed
+
+		-- Dragon
+		[3104] = {chance = 5000, newItem = 5877, after = 3105},
+		[5973] = {chance = 5000, newItem = 5877, after = 3105}, -- after being killed
+
+		-- Dragon Lord
+		[2881] = {chance = 5000, newItem = 5948, after = 2882},
+		[5984] = {chance = 5000, newItem = 5948, after = 2882}, -- after being killed
+
+		-- Behemoth
+		[2931] = {chance = 10000, newItem = 5893, after = 2932},
+		[5999] = {chance = 10000, newItem = 5893, after = 2932}, -- after being killed
+
+		-- Bone Beast
+		[3031] = {chance = 6000, newItem = 5925, after = 3032},
+		[6030] = {chance = 6000, newItem = 5925, after = 3032}, -- after being killed
+
+		-- Clomp
+		[25399] = {chance = 50000, newItem = 24842, after = 25400},
+		[25398] = {chance = 50000, newItem = 24842, after = 25400}, -- after being killed
+
+		-- Piece of Marble Rock
+		[11343] = {
+			{chance = 530, newItem = 11346, desc = "This little figurine of a goddess was masterfully sculpted by |PLAYERNAME|."},
+			{chance = 9600, newItem = 11345, desc = "This little figurine made by |PLAYERNAME| has some room for improvement."},
+			{chance = 24000, newItem = 11344, desc = "This shoddy work was made by |PLAYERNAME|."}
+		},
+
+		-- Ice Cube
+		[7441] = {chance = 22000, newItem = 7442},
+		[7442] = {chance = 4800, newItem = 7444},
+		[7444] = {chance = 900, newItem = 7445},
+		[7445] = {chance = 40, newItem = 7446}
+	},
+	[5942] = {
+		-- Demon
+		[2916] = {chance = 3000, newItem = 5906, after = 2917},
+		[5995] = {chance = 3000, newItem = 5906, after = 2917}, -- after being killed
+
+		-- Vampires
+		[2956] = {chance = 6000, newItem = 5905, after = 2957}, -- vampire
+		[6006] = {chance = 6000, newItem = 5905, after = 2957}, -- vampire, after being killed
+		[9654] = {chance = 6000, newItem = 5905, after = 9658}, -- vampire bride
+		[9660] = {chance = 6000, newItem = 5905, after = 9658}, -- vampire bride, after being killed
+		[21275] = {chance = 6000, newItem = 5905, after = 21276}, -- vampire viscount
+		[21278] = {chance = 6000, newItem = 5905, after = 21276} -- vampire viscount, after being killed
+	}
+}
+
+local skinning = Action()
+
+function skinning.onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	local skin = config[item.itemid][target.itemid]
+	if not skin then
+		player:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
+		return true
+	end
+
+	local randomChance = math.random(1, 100000)
+	local effect = CONST_ME_MAGIC_GREEN
+	local transform = true
+	if type(skin[1]) == "table" then
+		local added = false
+		local _skin
+		for i = 1, #skin do
```

Comment:
iterate with ipairs and remove _skin variable
for _, skinChild in ipairs(skin)

Reply:
Numeric loops are faster compared to pairs / ipairs
Ex https://medium.com/@RBX_Crazyman32/battle-of-the-loops-c001bcb4961c#1e2e

Reply:
quoting the thing you linked:
`A “numeric” loop is faster. However, in most applications, using a “generic” loop is fine. Unless you actually have performance issues, don’t worry about switching to “numeric” loops. `

Reply:
Yes both are fine but why change it? It won't make the code cleaner or faster etc
How ever I agree to change the variable name.

Reply:
sorry but my brain lacks the knowledge to do whatever you guys are suggesting xD I'd appreciate a pastebin with the change tho ;P

Reply:
```
		for _, skinChild in ipairs(skin)
			if randomChance <= skinChild.chance then
```

Then change all the _skin with skinChild 

If you ask me changing all the _skin to skinChild is enough.

Reply:
please check if it is correct now